metadata-ingestion/src/datahub/ingestion/source/powerbi/rest_api_wrapper/data_classes.py

```diff
@@ -0,0 +1,200 @@
+from dataclasses import dataclass
+from enum import Enum
+from typing import Any, Dict, List, Optional
+
+from datahub.emitter.mcp_builder import PlatformKey
+
+
+class WorkspaceKey(PlatformKey):
+    workspace: str
+
+
+@dataclass
+class Workspace:
+    """
+    PowerBi Workspace
+    """
+
+    id: str
+    name: str
+    dashboards: List["Dashboard"]
+    reports: List["Report"]
+    datasets: Dict[str, "PowerBIDataset"]
+    report_endorsements: Dict[str, List[str]]
+    dashboard_endorsements: Dict[str, List[str]]
+    scan_result: dict
+
+    def get_urn_part(self):
+        return self.name
+
+    def get_workspace_key(self, platform_name: str) -> PlatformKey:
+        return WorkspaceKey(
+            workspace=self.get_urn_part(),
+            platform=platform_name,
+        )
+
+
+@dataclass
+class DataSource:
+    """
+    PowerBi
```

Comment:
Is this comment supposed to have more text?

Reply:
removed